Bottom: b9adbe5040aa137ab095e164302cd1f5c5d3c034
Top:    438e4fc39c1bf607187bca6fd6f57145ed52945f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-27 20:35:06 +0100

DRAW_FUNCS: Attempt at different render order for PCB+GL


---

diff --git a/src/draw_funcs.c b/src/draw_funcs.c
index eb0cc57..45c3c82 100644
--- a/src/draw_funcs.c
+++ b/src/draw_funcs.c
@@ -433,6 +433,12 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
   int group = GetLayerGroupNumberByPointer (layer);
   struct poly_info info = {drawn_area, layer};
 
+  if (layer_num < max_copper_layer)
+    {
+      r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_hole_callback, NULL);
+      r_search (PCB->Data->via_tree, drawn_area, NULL, via_hole_callback, NULL);
+    }
+
   /* print the non-clearing polys */
   r_search (layer->polygon_tree, drawn_area, NULL, poly_callback, &info);
 
@@ -443,6 +449,13 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
   if (TEST_FLAG (CHECKPLANESFLAG, PCB))
     return;
 
+  if (layer_num < max_copper_layer)
+    {
+      r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_hole_callback, NULL);
+      r_search (PCB->Data->via_tree, drawn_area, NULL, via_hole_callback, NULL);
+    }
+
+  /* draw all visible lines this layer */
   r_search (layer->line_tree, drawn_area, NULL, line_callback, layer);
   r_search (layer->arc_tree,  drawn_area, NULL, arc_callback,  layer);
   r_search (layer->text_tree, drawn_area, NULL, text_callback, layer);
@@ -478,8 +491,6 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
 
   /* draw vias */
   r_search (PCB->Data->via_tree, drawn_area, NULL, via_inlayer_callback, layer);
-  r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_hole_callback, NULL);
-  r_search (PCB->Data->via_tree, drawn_area, NULL, via_hole_callback, NULL);
 }
 
 struct draw_funcs d_f = {
